Add notes about incompatibility between Pages and ACM (#7807)
cloudflare/cloudflare-docs: content/pages/platform/known-issues.md

## Custom Domains

- It is currently not possible to add a custom domain with
    - a wildcard, for example, `*.domain.com`.

>     - a Worker already routed on that domain.

Reviewer: Don't think this is working as intended
![image](https://user-images.githubusercontent.com/8492901/221968613-2cb2e31f-855b-45c5-a012-8a6a6ff7ba20.png)

Author: Oops, will have to check how that's done properly

- It is currently not possible to add a custom domain with a Cloudflare Access policy already enabled on that domain.

- Cloudflare's Load Balancer does not work with `*.pages.dev` projects; an `Error 1000: DNS points to prohibited IP` will appear.

- When adding a custom domain, the domain will not verify if Cloudflare cannot validate a request for an SSL certificate on that hostname. In order for the SSL to validate, ensure Cloudflare Access or a Cloudflare Worker is allowing requests to the validation path: `http://{domain_name}/.well-known/acme-challenge/*`.

> - [Advanced Certificates](/ssl/edge-certificates/advanced-certificate-manager/) are not able to be used with Cloudflare Pages due to [certificate prioritization](/ssl/reference/certificate-and-hostname-priority). This is because Pages uses Cloudflare for SaaS for Custom Domains.

Reviewer: ```suggestion
- [Advanced Certificates](/ssl/edge-certificates/advanced-certificate-manager/) are not able to be used with Cloudflare Pages due to SSL for SaaS's [certificate prioritization](/ssl/reference/certificate-and-hostname-priority).
```

## Custom Domains

- It is currently not possible to add a custom domain with
    
    - a wildcard, for example, `*.domain.com`.
    - a Worker already routed on that domain.

- It is currently not possible to add a custom domain with a Cloudflare Access policy already enabled on that domain.

- Cloudflare's Load Balancer does not work with `*.pages.dev` projects; an `Error 1000: DNS points to prohibited IP` will appear.

- When adding a custom domain, the domain will not verify if Cloudflare cannot validate a request for an SSL certificate on that hostname. In order for the SSL to validate, ensure Cloudflare Access or a Cloudflare Worker is allowing requests to the validation path: `http://{domain_name}/.well-known/acme-challenge/*`.

> - [Advanced Certificates](/ssl/edge-certificates/advanced-certificate-manager/) cannot be used with Cloudflare Pages due to SSL for SaaS's [certificate prioritization](/ssl/reference/certificate-and-hostname-priority).

Reviewer 2: ```suggestion
- [Advanced Certificates](/ssl/edge-certificates/advanced-certificate-manager/) cannot be used with Cloudflare Pages due to SSL for SaaS's [certificate prioritization](/ssl/reference/certificate-and-hostname-priority/).
```
